tests/synthdnn/matmul/primitive.py

```diff
@@ -0,0 +1,147 @@
+
+
+class Layouts:
+    class Layout:
+        def __init__(self, layout):
+            self.A, self.B, self.C = layout.split(":")
+
+        def benchdnn_str(self):
+            return f"--allow-enum-tags-only=0 --stag={self.A} --wtag={self.B} --dtag={self.C}"
+
+    def __init__(self, layouts, ndims):
+        if(layouts == "all"):
+            self.values = self.supported(ndims)
+        else:
+            self.values = [self.Layout(x) for x in layouts.split(",")]
+
+    def __iter__(self):
+        return iter(self.values)
+
+    @staticmethod
+    def supported(ndims):
+        if ndims < 2 or ndims > 6:
+            raise RuntimeError(f"No support for ndims={ndims}")
+        dims_base="abcdef"
+        perms = [''.join(p) for p in itertools.permutations(dims_base[:ndims])]
+        perms.insert(0, "any")
+        return [Layouts.Layout(f"{a}:{b}:{c}") for a,b,c in itertools.product(perms, perms, perms)]
+
+
+class Types:
+    class Type:
+        def __init__(self, type_str):
+            s = type_str.split("(")
+            self.A, self.B, self.C = s[0].split(":")
+            if len(s) < 2:
+                self.mode = None
+            else:
+                self.mode = s[1].strip(")")
+
+        def __str__(self):
+            mode_str = ""
+            if self.mode:
+                mode_str=f"({self.mode})"
+            return f"{self.A}:{self.B}:{self.C}{mode_str}"
+
+        def benchdnn_str(self):
+            mode_str = ""
+            if not self.mode is None:
+                mode_str = f"--attr-fpmath={self.mode}"
+            return f"--dt={self.A}:{self.B}:{self.C} {mode_str}"
+
+        def __eq__(self, other):
+            return (self.A, self.B, self.C, self.mode) == (
+                other.A,
+                other.B,
+                other.C,
+                other.mode,
+            )
+
+    def __init__(self, types):
+        if(types == "all"):
+            self.values = self.supported()
+        else:
+            self.values = [self.Type(x) for x in types.split(",")]
+
+    def __str__(self):
+        return ",".join([str(x) for x in self.values])
+
+    def __iter__(self):
+        return iter(self.values)
+
+    @staticmethod
+    def supported():
+        support_matrix =  [
+            [["f64"], ["f64"], ["f64"], [""]],
+            [["f32"], ["f32"], ["f32"], ["", "(tf32)","(f16)", "(bf16)"]],
+            [["f16"], ["f16"], ["f16", "u8", "s8"], [""]],
+            [["f16"], ["f16", "u8", "s8","u4","s4"], ["f16", "u8", "s8"], ["", "(f16:true)"]],
+            [["f16"], ["f16", "u8", "s8"], ["f32"], [""]],
+            [["bf16"], ["bf16"], ["bf16", "u8", "s8"], [""]],
+            [["f32", "bf16"], ["bf16", "u8", "s8","u4","s4"], ["bf16", "u8", "s8"], ["", "(bf16:true)"]],
+            [["bf16"], ["bf16", "u8", "s8"], ["f32"], [""]],
+            [["f32"], ["u8", "s8", "u4", "s4"], ["f32"], [""]],
+            [["u8","s8"], ["u8", "s8"], ["f32", "bf16", "f16", "s32", "u8", "s8"], [""]],
```

Comment:
The [documentation's support matrix](https://oneapi-src.github.io/oneDNN/dev_guide_matmul.html#data-types) differs from these lines quite a bit. Is there a reason not to effectively copy the intended support into this table?

Reply:
I was looking at bf16 mainly:
![image](https://github.com/user-attachments/assets/007bde0b-318f-497a-a17e-79b9888cdae1)

Reply:
Hardware support for u8 weights appeared with AMX only, so it's not advertised.

Reply:
Thanks for the information @dzarukin, I will go ahead and drop that combination.

@Simonsays095, thanks for pointing that out, I believe I have closed that gap. The testing set still isn't quite aligned as I am attempting to match the documentation with what I have seen in practice.  In particular, I have not seen any mixed bf16/f16 workloads, so I am dropping those cases. In addition, the f16 and bf16 data types are used in similar scenarios, so I aligned their coverage. We can change this if there are disagreements with the approach.